src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts

```diff
@@ -0,0 +1,45 @@
+import { MutableRefObject } from 'react'
+
+import { getIsRequestAborted, post } from '@Common/Api'
+import { ROUTES } from '@Common/Constants'
+import { getUrlWithSearchParams, showError } from '@Common/Helper'
+import { UploadFileDTO } from '@Shared/types'
+
+export const uploadCDPipelineFile = async ({
+    file,
+    appId,
+    envId,
+    allowedExtensions,
+    maxUploadSize,
+    abortControllerRef,
+}: {
+    file: File[]
+    appId: number
+    envId: number
+    allowedExtensions?: string[]
+    maxUploadSize?: number
+    abortControllerRef?: MutableRefObject<AbortController>
+}): Promise<UploadFileDTO> => {
```

Comment:
- allowedExtensions, maxUploadSize are already in one of the types. pls reuse
- abortControllerRef should be read from APIOptions

Reply:
abortControllerRef part seems to be left